rolling-shutter/keyper/keyper.go

```diff
@@ -246,21 +253,22 @@ func (kpr *keyper) handleOnChainChanges(ctx context.Context, tx pgx.Tx) error {
 // enough keypers have seen a block past the start block of some BatchConfig. We only send messages
 // when the current block we see, could lead to a batch config being started.
 func (kpr *keyper) sendNewBlockSeen(ctx context.Context, tx pgx.Tx) error {
-	qc := commondb.New(tx)
 	q := kprdb.New(tx)
 	lastBlock, err := q.GetLastBlockSeen(ctx)
 	if err != nil {
 		return err
 	}
-	nextBlock, err := qc.GetNextBlockNumber(ctx)
+	currentBlock, err := medley.Retry(ctx, func() (uint64, error) {
+		return kpr.l1Client.BlockNumber(ctx)
```

Comment:
This is running inside an active database transaction. We should not do any HTTP requests during that.

Reply:
fixed